src/hooks/formHooks/workout/useNewWorkoutForm.tsx

```diff
@@ -120,10 +117,58 @@ export const useNewWorkoutForm = ({
     ]
   );
 
+  const onDraftSave = useCallback(async () => {
+    setPending(true);
+    clearErrors();
+    const formData: AddWorkoutForm = getValues();
+    try {
+      // validate for draft save
+      await draftSchema.validate(formData, { abortEarly: false });
+      // add draft workout
+      const newDraftWorkout = generateNewWorkout(formData, user!, true);
+
+      // todo: refactory when backend will be written. Delete addQueryUserProgress because it will be done in backend in addNewWorkout route
+      addQueryWorkout(newDraftWorkout)
+        .then(() => {
+          resetForm();
+
+          const newUserProgress = generateUserProgress(newDraftWorkout);
+          return newUserProgress.map((userProgres) =>
+            addQueryUserProgres(userProgres)
+          );
+        })
+        .finally(() => {
+          setPending(false);
+          setIsDraftSubmited(true);
+        });
+    } catch (error: unknown) {
+      // ? question -  do i have to handle else ?
```

Comment:
masz na myśli inne typu errorów? Wystarczy że zwrócisz ten error, jeśli będzie potrzeba handlowania go to się go doda, ale jeśli jest error to chcemy żeby był zwracany, tak jak zawsze jest setPending(false);